api/v1alpha1/authorization_types.go

```diff
@@ -72,6 +96,36 @@ type Principal struct {
 	// JWT authentication in the same `SecurityPolicy`.
 	// +optional
 	JWT *JWTPrincipal `json:"jwt,omitempty"`
+
+	// Headers authorize the request based on user identity extracted from custom headers.
+	// If multiple headers are specified, all headers must match for the rule to match.
+	//
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=256
+	// +notImplementedHide
+	Headers []AuthorizationHeaderMatch `json:"headers,omitempty"`
```

Comment:
A new `AuthorizationHeaderMatch` is introduced for authorization, as the existing `HeaderMatch` is not suitable for this use case.